src/app/main.cpp

```diff
@@ -318,17 +316,13 @@ int main(int argc, char *argv[])
 void reportToUser(const char *str)
 {
     const size_t strLen = strlen(str);
-#ifndef Q_OS_WIN
-    if (write(STDERR_FILENO, str, strLen) < static_cast<ssize_t>(strLen))
-    {
-        const auto dummy = write(STDOUT_FILENO, str, strLen);
+#ifdef Q_OS_WIN
+    if (_write(STDERR_FILENO, str, strLen) < static_cast<int>(strLen))
+        Q_UNUSED(_write(STDOUT_FILENO, str, strLen));
```

Comment:
Why do we even need Q_UNUSED here?

Reply:
I think it was used to suppress "unused return value" compiler warnings.

Reply:
IIRC, Q_UNUSED is intended to be used with function parameters. Won't it have an undesirable side effect in this case (e.g., skipping a function call by the compiler)?

Reply:
>Won't it have an undesirable side effect in this case (e.g., skipping a function call by the compiler)?

By c++ definition, the function parameters are evaluated before going into the function itself. So the `_write()` is still executed.

Reply:
Some alternatives are:
* https://en.cppreference.com/w/cpp/language/attributes/maybe_unused
* https://en.cppreference.com/w/cpp/utility/tuple/ignore

I can switch to them if you find them any better.

Reply:
>Q_UNUSED isn't a function.

True. However look up the definition, it is just applying a simple cast.

>I worry that this may have the same effect as Q_ASSERT(_write(STDOUT_FILENO, str, strLen) > 0); does under some conditions.

If that is true, then `Q_UNUSED` would have no effect on stopping compiler warnings and broken a lot of code...

>std::ignore should work.

Updated to use it.